llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -4080,6 +4080,17 @@ SDValue AMDGPUTargetLowering::performMulCombine(SDNode *N,
   SDValue N0 = N->getOperand(0);
   SDValue N1 = N->getOperand(1);
 
+  // FIXME: Probably should only be done for gfx9 and onward.
```

Comment:
Yes should check for gfx9-insts. We have these wrappers:

```
  bool hasSwap() const {
    return GFX9Insts;
  }

  bool hasScalarPackInsts() const {
    return GFX9Insts;
  }

  bool hasScalarMulHiInsts() const {
    return GFX9Insts;
  }


```

Should have another for the shl add

Reply:
Add a DAG comment and explain why it's worth doing this